core/amber/src/main/protobuf/edu/uci/ics/amber/engine/architecture/worker/controlreturns.proto

```diff
@@ -18,13 +18,27 @@ message ControlException {
     string msg = 1;
 }
 
+message TypedValue {
+    string expression = 1;
+    string value_ref = 2;
+    string value_str = 3;
+    string value_type = 4;
+    bool expandable = 5;
```

Comment:
based on the code and the discussion, we are modeling the value similar to what's in json, primitive type (expandable = false), or dict-like type (expandable = true), i believe you treat array-like type also like dict-like type (expandable = true). 
I think this design is good. Can we add a description somewhere to explain our value model? 

Also I'm just curirous that are there benefits to treat dict-like type and array-like type differenetly. I think it's fine to treat them as both dict-like, because Javascript does it this way.

Reply:
added doc